src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -24,7 +24,7 @@ public static bool IsInvalid(this IOperation operation, Compilation compilation,
 
             // if wrong compilation is given, GetSemanticModel will throw due to tree not belong to the given compilation.
             var model = compilation.GetSemanticModel(operation.Syntax.SyntaxTree);
-            return model.GetDiagnostics(operation.Syntax.Span, cancellationToken).Any(d => d.Severity == DiagnosticSeverity.Error);
+            return model.GetDiagnostics(operation.Syntax.Span, cancellationToken).Any(d => d.DefaultSeverity == DiagnosticSeverity.Error);
```

Comment:
So even if I have an error in code (because I have set /warnaserror), this method will return false?

Also if we are returning true only for errors, then the method should be called HasErrors, not HasDiagnostics.